core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+            ticket_name: str. The name assigned to the ticket on creation.
+
+        Returns:
+            str. The generated ID for this entity using the current datetime in
+            seconds (as the entity's creation timestamp), a SHA1 hash of the
+            ticket_name, and a random string, of the form
+            '[creation_datetime]:[hash(ticket_name)]:[random hash]'.
+        """
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            name_hash = utils.convert_to_hash(
+                ticket_name, base_models.ID_LENGTH)
+            random_hash = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            new_id = '%s.%s.%s' % (
+                datetime.datetime.utcnow().second, name_hash, random_hash)
+            if not cls.get_by_id(new_id):
+                return new_id
+        raise Exception(
+            'The id generator for AppFeedbackReportTicketModel is producing too'
+            'many collisions.')
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model doesn't contain any information directly corresponding to a
+        user.
+        """
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model doesn't contain any data directly corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'ticket_name': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'github_issue_number': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'is_archived': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'newest_report_timestamp': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'report_ids': base_models.EXPORT_POLICY.NOT_APPLICABLE
+        })
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model doesn't contain any user data."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @staticmethod
+    def get_lowest_supported_role():
+        """The lowest supported role for feedback report tickets will be
+        moderator.
+        """
+        return feconf.ROLE_ID_MODERATOR
+
+
+class AppFeedbackReportStatsModel(base_models.BaseModel):
+    """Model for storing aggregate report stats on the tickets created.
+
+    Instances of this model contain statistics for different report types based
+    on the ticket they are assigned to and the date of the aggregation is on.
+
+    The id of each model instance is calculated by concatenating the platform,
+    ticket ID, and the date (in isoformat) this entity is tracking stats for.
+    """
+
+    # The unique ticket ID that this entity is aggregating for.
+    ticket_id = datastore_services.StringProperty(required=True, indexed=True)
+    # The platform that these statistics are for.
+    platform = datastore_services.TextProperty(
```

Comment:
Surely this is short enough to be a TextProperty?

Also, you might as well index it.

Reply:
I had used TextProperty since StringProperty can't be indexed=False, will change to StringProperty and indexed=True